### PR TITLE
Exogenous transition path for shipping, Steel, and Aluminum production 

### PR DESCRIPTION
**Pull request by martavp**
Transformation of shipping demand, as well as production of Steel and Aluminum, is currently assumed to occur overnight.
(i.e. is only represented by one value in the config.yaml file). This PR enables the use of exogenously defined transition paths for these transformations.

### CODE REVIEW

**Comment by martavp** [comment]
I just realized that I missed adding the oil consumed in shipping as a load to the EU oil bus.

File: scripts/prepare_sector_network.py [hunk 1]
```diff
@@ -1726,6 +1727,27 @@ def add_industry(n, costs):
...
+
+        shipping_oil_share = 1 - shipping_hydrogen_share
+        
+        n.madd("Load",
+               nodes,
+               suffix=" shipping oil",
+               bus="EU oil",
+               carrier="shipping oil",
+               p_set=shipping_oil_share*nodal_energy_totals.loc[nodes,["total international navigation","total domestic navigation"]].sum(axis=1)*1e6/8760.
+               )
```
Comment (fneum) [inline scripts/prepare_sector_network.py:1740]:
```suggestion
        p_set = shipping_oil_share * nodal_energy_totals.loc[nodes, all_navigation].sum(axis=1) * 1e6 / 8760.
        n.madd("Load",
               nodes,
               suffix=" shipping oil",
               bus="EU oil",
               carrier="shipping oil",
               p_set=p_set
)
```

Comment (martavp) [reply scripts/prepare_sector_network.py]: Great, it is clearer as you propose.

**Comment by lisazeyen** [comment]
That is really cool! I guess n the Snakefile there is the wildcard `planning_horizons` missing for the benchmarks see [here](https://github.com/PyPSA/pypsa-eur-sec/pull/149/commits/2e336e5e7043976c97580471cbd6075d837bf742). I made a new pull request based on this branch to include today's district heating share and the options to increase it during the planning horizon.

**Comment by fneum** [comment]
> That is really cool! I guess n the Snakefile there is the wildcard `planning_horizons` missing for the benchmarks see [here](https://github.com/PyPSA/pypsa-eur-sec/pull/149/commits/2e336e5e7043976c97580471cbd6075d837bf742). I made a new pull request based on this branch to include today's district heating share and the options to increase it during the planning horizon.

That was my mistake when I merged the master.

**Review by fneum — APPROVED** [review]

**Comment by fneum** [comment]
Great, flexibly switching between exogenous transition paths and fixed choices for parameters is very nicely implemented and we can surely use this as role model for other parameters, too.

**Pull request merged**